axelrod/tests/strategies/test_zero_determinant.py

```diff
@@ -0,0 +1,303 @@
+"""Tests for the Zero Determinant strategies."""
+
+import unittest
+
+import axelrod
+from axelrod.strategies.zero_determinant import LRPlayer
+from .test_player import TestPlayer, test_four_vector
+
+C, D = axelrod.Action.C, axelrod.Action.D
+
+
+class TestLRPlayer(unittest.TestCase):
+
+    def test_exception(self):
+        player = LRPlayer()
+        with self.assertRaises(ValueError):
+            player.receive_match_attributes(0, 0, -float("inf"))
+
+
+class TestZDExtortion(TestPlayer):
+
+    name = "ZD-Extortion: 0.2, 0.1, 1"
+    player = axelrod.ZDExtortion
+    expected_classifier = {
+        'memory_depth': 1,
+        'stochastic': True,
+        'makes_use_of': set(["game"]),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def test_four_vector(self):
+        expected_dictionary = {(C, C): 0.64, (C, D): 0.18, (D, C): 0.28,
```

Comment:
Only to avoid the numerical manipulation carried out by the strategy to throw out numerical floating point errors at some point? (if it never fails then I'm completely fine with them being numeric, just trying to avoid a trivial fix in the future)

```
>>> import sympy as sym
>>> sym.nsimplify(.64), sym.nsimplify(0.18), sym.nsimplify(0.28)
(16/25, 9/50, 7/25)
```